app/components/page-components/Chatbot/Commands/ChatbotCustomCommands.vue.ts

```diff
@@ -0,0 +1,55 @@
+import ChatbotBase from 'components/page-components/Chatbot/ChatbotBase.vue';
+import { Component } from 'vue-property-decorator';
+import {
+  CustomCommandsResponse,
+  CustomCommandsData,
+  Pagination,
+  ChatbotAPIPutResponse
+} from 'services/chatbot/chatbot-interfaces';
+
+
+@Component({})
+export default class ChatbotDefaultCommands extends ChatbotBase {
+  commands: CustomCommandsData = [];
+  pagination: Pagination = {
+    current: 1,
+    total: 1
+  };
+
+  mounted() {
+    //
+    // get list of user's custom commands
+    //
+    this.fetchCommands(this.pagination.current);
+  }
+
+  fetchCommands(page: number) {
+    // fetch custom commands
+    this.chatbotApiService
+      .fetchCustomCommands(page)
+      .then((response: CustomCommandsResponse) => {
+        console.log(response.data);
+        this.commands = response.data;
+        this.pagination = response.pagination;
+      });
+  }
+
+  openCommandWindow() {
+    this.chatbotCommonService.openCommandWindow();
+  }
+
+  toggleEnableCommand(commandId: string, index: number, isEnabled: boolean) {
+    const commandToBeUpdated = this.commands[index];
+
+    this.chatbotApiService
+      .updateCustomCommand(commandId, {
+        ...commandToBeUpdated,
+        enabled: isEnabled
+      })
+      .then((response: ChatbotAPIPutResponse) => {
+        if (response.success) {
+          this.commands[index].enabled = isEnabled;
```

Comment:
Am I right you use `index` for front-end only? If you have a relation like  `1 to 1` between `commandId` and `index` than `index` it's an excess argument here. Just get it in the function body
```
const index = this.command.findIndex(cmd => cmd.id == commandId);
```

Reply:
yeah, its from frontend only, I just thought findIndex is uncessary if I have the index from frontend, but regardless I am removing this entire and just fetching GET all commands to keep data consistent.